src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RemoteProjectSnapshotProjectEngineFactory.cs

```diff
@@ -14,16 +15,23 @@ internal class RemoteProjectSnapshotProjectEngineFactory : DefaultProjectSnapsho
         public static readonly IFallbackProjectEngineFactory FallbackProjectEngineFactory = new FallbackProjectEngineFactory();
 
         private readonly FilePathNormalizer _filePathNormalizer;
+        private readonly IOptionsMonitor<RazorLSPOptions> _optionsMonitor;
 
-        public RemoteProjectSnapshotProjectEngineFactory(FilePathNormalizer filePathNormalizer) : 
+        public RemoteProjectSnapshotProjectEngineFactory(FilePathNormalizer filePathNormalizer, IOptionsMonitor<RazorLSPOptions> optionsMonitor) : 
```

Comment:
Instead of taking in the monitor, if this is re-constructed every single time could we take in the `RazorLSPOptions` themselves? This way subsquent parses with the "engine" don't generate different results

Reply:
From debugging, it seems like the `RemoteProjectSnapshotProjectEngineFactory` constructor is only called once? So if we pass in the `RazorLSPOptions` themselves, I don't think there would be any way of updating them later.

Reply:
oh interesting, Ajay may not have done all the hookup to cause re-parses when settings change. That being said a change like that is probably much larger to make and would require some extensions to our ProjectSnapshot based system. For another day 😄